src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -223,6 +223,20 @@ internal void ClearAnalyzerExceptionHandlers(DiagnosticAnalyzer analyzer)
             }
         }
 
+        internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diagnostic, AnalyzerExecutor analyzerExecutor)
+        {
+            var supportedDescriptors = GetSupportedDiagnosticDescriptors(analyzer, analyzerExecutor);
+            foreach (var descriptor in supportedDescriptors)
```

Comment:
this is iterating through supported descriptions, can we think about a bit more performant way to do this at right position?